### PR TITLE
Solve problem saving empty swatches in admin

### PR DESCRIPTION
**Pull request by enriquei4**
### Description
Solve problem saving empty swatches in admin remving html ement instead of add hidden.

### Fixed Issues (if relevant)

1. Swatch Attribute is not getting save while deleting a swatch row with empty admin scope text #13117

### Manual testing scenarios

1. Login to admin and Goto Store- >Attribute->Product
2. Find a attribute like color
3. Open color attribute and make it to Visual Swatch type
4. In Manage Swatches section add color swatches
5. Now Click on Add Swatch button, a row will appear just remove it without filling anything into it.
6. Now is saved. Before it throws: The value of Admin scope can't be empty.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by enriquei4** [comment]
@serhii-balko i commited with a different email, i created this new PR to clean: https://github.com/magento/magento2/pull/13154

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php [hunk 1]
```diff
@@ -110,6 +110,11 @@ public function execute()
                 $options
             );
             $valueOptions = (isset($options['value']) && is_array($options['value'])) ? $options['value'] : [];
+            foreach ($valueOptions as $key => $valueOption) {
+                if (isset($options['delete'][$key]) && $options['delete'][$key]) {
```
Comment (orlangur) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: Better use `if(!empty($options['delete'][$key]))`.

**Review by serhii-balko — COMMENTED** [review]
You should also add a new test cases with empty 'deleted' options in  https://github.com/magento/magento2ce/blob/2.3-develop/app/code/Magento/Catalog/Test/Unit/Controller/Adminhtml/Product/Attribute/ValidateTest.php#L325

**Comment by enriquei4** [comment]
Ok i'm working on the Test.

**Review by magento-engcom-team — APPROVED** [review]

**Review by serhii-balko — APPROVED** [review]

**Comment by enriquei4** [comment]
Hi guys, i fixed the dataprovider it had a mistake, now is solved.  $isError should be "false"

File: app/code/Magento/Catalog/Test/Unit/Controller/Adminhtml/Product/Attribute/ValidateTest.php [hunk 2]
```diff
@@ -249,6 +249,20 @@ public function provideUniqueData()
...
+                        "option_1" => [2, 0],
+                        "option_2" => ["", ""],
+                    ],
+                    'delete' => [
+                        "option_0" => "",
+                        "option_1" => "",
+                        "option_2" => "1",
+                    ]
+                ], false
+            ],
```
Comment (serhii-balko) [inline app/code/Magento/Catalog/Test/Unit/Controller/Adminhtml/Product/Attribute/ValidateTest.php]: result param should be like this
```
(object) [
    'error' => false,
]
```

**Review by serhii-balko — APPROVED** [review]

**Review by serhii-balko — COMMENTED** [review]
the best way to test this case: add new cases in  `provideEmptyOption` provider like this
```
'empty admin scope options and deleted' => [
                [
                    'value' => [
                        "option_0" => [''],
                    ],
                    'delete' => [
                        'option_0' => '1',
                    ],
                ],
                (object) [
                    'error' => false,
                ],
            ],
            'empty admin scope options and not deleted' => [
                [
                    'value' => [
                        "option_0" => [''],
                    ],
                    'delete' => [
                        'option_0' => '0',
                    ],
                ],
                (object) [
                    'error' => true,
                    'message' => 'The value of Admin scope can\'t be empty.',
                ],
            ],
```

**Comment by enriquei4** [comment]
Hi @serhii-balko should I remove the previous test case for testUniqueValidation?

**Review by serhii-balko — COMMENTED** [review]
Yes,  it will be better if you remove it

**Comment by enriquei4** [comment]
But that (first)case is working well, and is testing well the code. If i remove the new code (validate.php) the test fails, then i think that is useful.

**Comment by enriquei4** [comment]
@serhii-balko did you read my last comment :) (I understand that you could be busy)

**Review by magento-engcom-team — COMMENTED** [review]
Yes, you are right, We approved this changes
